src/Microsoft.HttpRepl/Preferences/IPreferencesProvider.cs

```diff
@@ -0,0 +1,12 @@
+using System.Collections.Generic;
+
+namespace Microsoft.HttpRepl.Preferences
+{
+    public interface IPreferencesProvider
```

Comment:
Not a fan of the name here (or of the implementing class) so open to suggestions.

Reply:
How do you like just calling this IPreferences?

Reply:
I certainly like that more, though calling the default implementation of it Preferences leads to lots of fun namespace/class collision issues. Not sure I like the burden that added.

Reply:
yeah, I see what you mean about the name.  The default could be something like UserFolderPreferences or something to indicate what's unique to that implementation.

Reply:
Works for me. Done.